filter_plugins/cisco_nxos.py

```diff
@@ -0,0 +1,444 @@
+
+        # "speed" field
+        if "speed" in fields:
+            config_dict[dict_key].append(f"speed {fields['speed']}")
+
+        # "fec" field
+        if "fec" in fields:
+            if fields["fec"]:
+                config_dict[dict_key].append("fec auto")
+            else:
+                config_dict[dict_key].append("fec off")
+
+        # Set to L3 mode if needed
+        is_l3 = "ip4" in fields or "ip6" in fields
+        if is_l3:
+            config_dict[dict_key].append("no switchport")
+
+        # "ip4" field
+        if "ip4" in fields:
+            config_dict[dict_key].append(f"ip address {fields['ip4']}")
+
+        # "ip6" field
+        if "ip6" in fields:
+            config_dict[dict_key].append(f"ipv6 address {fields['ip6']}")
+
+        # "stp" field
+        if "stp" in fields:
+            stp_fields = fields["stp"]
+            if "edgeport" in stp_fields and stp_fields["edgeport"]:
+                edge_str = "spanning-tree port type edge"
+                if "bpduguard" in stp_fields and stp_fields["bpduguard"]:
+                    edge_str += " bpduguard"
+
+                config_dict[dict_key].append(edge_str)
+
+            if "rootguard" in stp_fields and stp_fields["rootguard"] and not is_l3:
+                config_dict[dict_key].append("spanning-tree guard root")
+
+            if "loopguard" in stp_fields and stp_fields["loopguard"]:
+                if is_l3:
+                    config_dict[dict_key].append("spanning-tree loopguard")
+                else:
+                    config_dict[dict_key].append("spanning-tree guard loop")
+
+        # "portmode" field
+        if "portmode" in fields:
+            portmode_str = ""
+            if fields["portmode"] == "access":
+                portmode_str = "switchport mode access"
+            elif fields["portmode"] == "trunk" or fields["portmode"] == "hybrid":
+                portmode_str = "switchport mode trunk"
+
+            config_dict[dict_key].append(portmode_str)
+
+            if is_portchannel:
+                for member_port in getLAGMembers(fields):
+                    if member_port not in config_dict:
+                        config_dict[member_port] = []
+
+                    config_dict[member_port].append(portmode_str)
+
+        # "tagged" field
+        if "tagged" in fields:
+            allowed_vl_string = ",".join(list(map(str, fields["tagged"])))
+            tagged_str = f"switchport trunk allowed vlan {allowed_vl_string}"
+            config_dict[dict_key].append(tagged_str)
+
+            if is_portchannel:
+                for member_port in getLAGMembers(fields):
+                    if member_port not in config_dict:
+                        config_dict[member_port] = []
+
+                    config_dict[member_port].append(tagged_str)
+
+        # "untagged" field
+        if "untagged" in fields:
+            untagged_str = ""
+            if fields["portmode"] == "hybrid":
```

Comment:
in [line 307](https://github.com/CCI-MOC/ansible-switches/pull/59/files#diff-dd31f0cdd8a2ec6ba878bc26867abc7a3e3e1f6ba03d919eb9c7ab147990d1ecR307) we allow portmode to be either `trunk` or `hybrid` but here we expect it be only `hybrid`. So, if the manifests said "trunk", my switchport will be put in trunk mode but then this will not set the trunk native vlan and instead it'll set access vlan which would do nothing.

Reply:
sorry I still don't understand what you mean here. My question is why can't we do this:

```python
if "untagged" in fields:
    untagged_str = ""
    if fields["portmode"] in ["hybrid", "trunk"]:
        untagged_str = f"switchport trunk native vlan {fields['untagged']}"
    elif fields["portmode"] == "access":
        untagged_str = f"switchport access vlan {fields['untagged']}"
```


so we effectively treat hybrid as trunk

Reply:
Because I don't want portmode "trunk" and "untagged" to be a valid schema supplied together. It's not a valid schema for OS9 so I wanted to go with the minimum there. In other words the manifests are validated agnostic of the switch OS, so the validation needs to work for every switch OS

Reply:
ah, so OS9 expects only tagged in trunk mode. I guess we will deal with this later then and it's fine for now.